### DISCUSSION TITLE
[Info]: Fontawesome v6 icon name changes

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#3575] [General]
This is a sort of PSA (I assume).

With Fontawesome v6 came a lot of icon renames. Depending on where or for what you used the icons would this have minor inconveniences (i.e. `:fontawesome-<type>-<icon>:` not rendering) to [complete breaking of your deployment/build process](https://app.netlify.com/sites/andres-blog/deploys/6206bd2999fedd3f28ce07a0#L196) (Update: I no longer use netlify for my personal pages, so this link won't work anymore).

This affects anyone that uses the latest Material for MkDocs release (I think. Or is it related to the emoji extension?) and the best way to fix this is by renaming any appearance of old icon names in the `mkdocs.yml`, any Theme extensions you have and the markdown files themself.

Fontawesome provides an entire list of all changed icon names in v6:
https://fontawesome.com/docs/web/setup/upgrade/whats-changed#icons-renamed-in-version-6

----

Just to be safe is here the obligatory message that this is not an official message from Fontawesome. I do not work for them nor claim to do.
Any info is provided as is without guarantee, bla bla bla, you know the drill...

**Comment by squidfunk** [comment]
That's correct. I was not aware that FontAwesome 6 was such a breaking change, which is why I slipped it in a patch release. I've added a note retroactively to the changelog. If you set up configuration validation as described in the [creating your site](https://squidfunk.github.io/mkdocs-material/creating-your-site/#configuration) guide, it will at least tell you which icons are no longer valid in `mkdocs.yml`.

If somebody manages to provide a script to automatically rename all icons in Markdown files, feel free to share it here.

**Reply by SohamJ** [reply]
Not too fancy, but I used this for my use case which only uses a subset of solid and regular icons

The attached file copies the entire table from the fonstawesome site. I massaged each line to do generate a sed command for each icon update

> Note: only the Mac one is tested

```shell
# mac
cat fontawesome-changes.txt| awk '{ print "sed -E -i \x27\x27 \x27s/:fontawesome-(solid|regular|light|thin)-"$1":/:fontawesome-\\1-"$2":/g\x27 docs/**/*.md"}' > mass-icon-change.sh

# linux, which does not need single quotes after sed -i
cat fontawesome-changes.txt| awk '{ print "sed -E -i  \x27s/:fontawesome-(solid|regular|light|thin)-"$1":/:fontawesome-\\1-"$2":/g\x27 docs/**/*.md"}' > mass-icon-change.sh
```

Copy this script to the root folder of mkdocs project (assumes all md files under docs. That can be changed

```shell
chmod +x mass-icon-change.sh
cp mass-icon-change.sh /to/my/mkdocs/root/folder
./mass-icon-change.sh
```

Suggestions/fixes welcomed in case I missed anything

[fontawesome-changes.txt](https://github.com/squidfunk/mkdocs-material/files/8216524/fontawesome-changes.txt)
